### ISSUE TITLE
Debugger support on iOS

### ISSUE THREAD

**Issue by ds20211213** [ds20211213/frida-ceserver#4]
config.ini
```
{
  "target": "",
  "targetOS": 2,
  "mode": 1,
  "arch": 1,
  "fix_module_size": false,
  "ceversion": "7.4",
  "manualParser": false,
  "javaDissect": false,
  "native_ceserver_ip": "192.168.x.x:52734",
  "debugserver_ip": "192.168.x.x:1234",
  "custom_symbol_loader": ""
}
```

1. arch must be 1. This must be set to x64 because Cheat Engine does not support arm64.
2. Configure BinUtils and set disassembler to arm64.
3. Run debugserver on iOS.
```
./debugserver 0.0.0.0:1234
```
4. native_ceserver must also be run. This is because frida's own memory read/write must hit a breakpoint.

Only arm64 devices are supported.
**Currently only 「find out what writes to this addrss」 and「find out what accesses this address」are valid.**
Have fun!

![](https://user-images.githubusercontent.com/96031346/163714402-dfc8547f-8aa7-4b7e-b895-08778bee8b74.png)

**Comment by LanBaiCode** [comment]
![image](https://user-images.githubusercontent.com/72244576/163763274-47fadd9f-7986-4ceb-be6f-bda538876bc3.png)
No response ,[find out what accesses this address]

**Comment by ds20211213** [comment]
Thank you for your comment.
Did you find the referenced process itself?

**Comment by LanBaiCode** [comment]
> Thank you for your comment. Did you find the referenced process itself?

The game process that needs to be modified has been found, and the address can be added and the value in the memory can be modified
But can't use lldb

**Comment by ds20211213** [comment]
There was a problem with the exclusion control.
I would appreciate it if you could try again with a new commit.
Perhaps this will work.

**Comment by LanBaiCode** [comment]
> There was a problem with the exclusion control. I would appreciate it if you could try again with a new commit. Perhaps this will work.

yes it would be better if you have telegram so i can reply right away

**Comment by ds20211213** [comment]
> > There was a problem with the exclusion control. I would appreciate it if you could try again with a new commit. Perhaps this will work.
> 
> yes it would be better if you have telegram so i can reply right away

I am not familiar with telegram, but here is my account.
@doranekosystems

**Comment by Thekidiacs** [comment]
Wow this is awesome. I'm going to test this right now!

**Comment by Thekidiacs** [comment]
So I tried it and it was showing results when I did what writes/accesses the address but I'm pretty sure it wasn't the correct assembly. I'm using an iPad Air 4th generation on iOS 14.2 it is arm64e. It wouldn't let me search anymore after attaching the debugger. Maybe you have a solution? Here is what it looked like.
![Cheat Engine 7 2 4_19_2022 1_31_14 AM](https://user-images.githubusercontent.com/85626950/163936847-4a769f1a-fae7-4076-8d39-c5c587265e13.png)

**Comment by ds20211213** [comment]
Thekidacs

Please try with the latest commits.
https://github.com/ds20211213/frida-ceserver/commit/c6c53659faf7ff4a443d35b878b68caac961b793

Currently, cheat engine does not support arm64, so binutils configuration is required.
https://github.com/ds20211213/frida-ceserver#binutils

Also, native_ceserver needs to work.
This is because frida cannot read memory when the process is stopped in the debugger.
https://github.com/ds20211213/ceserver-ios-mini

**Comment by Thekidiacs** [comment]
is native_ceserver the ceserver that I put at /usr/bin? If so I am already using that. But am I still supposed to run the command python main.py "App name". I forgot to do binutils configuration so I will try that and report back.

**Comment by Thekidiacs** [comment]
So in cheat engine I went to memory view and went to "View" and couldn't find binutils?

**Comment by ds20211213** [comment]
Are you on telegram or wechat?

**Comment by Thekidiacs** [comment]
Yes, I use telegram whats your telegram username I'll add you.

**Comment by LanBaiCode** [comment]
> 传教士
> 
> 你是用电报还是微信？



> Thekidiacs
> 
> Are you on telegram or wechat?

I think now I can solve his problem

**Comment by LanBaiCode** [comment]
> Thekidiacs
> 
> Are you on telegram or wechat?

I think now I can solve his problem

**Comment by ds20211213** [comment]
telegram
@doranekosystems

**Comment by Thekidiacs** [comment]
Thanks @ds20211213. @LanBaiCode Thank you for willing to help me. What are your ideas?

**Comment by LanBaiCode** [comment]
> Thanks @ds20211213. @LanBaiCode Thank you for willing to help me. What are your ideas?

@ds20211213 He will help you solve the problem, the problem you encountered, I also asked him yesterday, and it has been successfully solved

**Comment by Thekidiacs** [comment]
Okay awesome. :)

**Comment by DoranekoSystems** [comment]
Supported update of ce itself.
arm64 disassembly and debugging is available without using binutils.
If you can build the ce itself, please try it if you like.

```
{
  "target": "",
  "targetOS": "ios",
  "mode": "attach",
  "arch": 3,
  "fix_module_size": false,
  "ceversion": "7.4",
  "manualParser": false,
  "javaDissect": false,
  "native_ceserver_ip": "192.168.x.x:52734",
  "debugserver_ip": "192.168.x.x:1234",
  "custom_symbol_loader": ""
}
```
![img](https://user-images.githubusercontent.com/96031346/164975214-bc0f0ca7-2c67-4d4e-a47a-92d6dc4b3fd2.png)

**Comment by yazigegeda** [comment]
> Supported update of ce itself. arm64 disassembly and debugging is available without using binutils. If you can build the ce itself, please try it if you like.
> 
> ```
> {
>   "target": "",
>   "targetOS": "ios",
>   "mode": "attach",
>   "arch": 3,
>   "fix_module_size": false,
>   "ceversion": "7.4.2",
>   "manualParser": false,
>   "javaDissect": false,
>   "frida_server_ip":"",
>   "native_ceserver_ip": "192.168.x.x:52734",
>   "debugserver_ip": "192.168.x.x:1234",
>   "custom_symbol_loader": ""
> }
> ```
> 
> ![img](https://user-images.githubusercontent.com/96031346/164975214-bc0f0ca7-2c67-4d4e-a47a-92d6dc4b3fd2.png)



targetOS  Does it support android?

**Comment by yazigegeda** [comment]
![1](https://user-images.githubusercontent.com/63853093/169647686-6197f8c9-ebe5-4be9-b065-eec3a8f0b2f0.jpg)
![2](https://user-images.githubusercontent.com/63853093/169647693-4c7607eb-a418-4a79-aaa2-ddefbf906db8.jpg)

android cannot find/access data

**Comment by DoranekoSystems** [comment]
> ![1](https://user-images.githubusercontent.com/63853093/169647686-6197f8c9-ebe5-4be9-b065-eec3a8f0b2f0.jpg) ![2](https://user-images.githubusercontent.com/63853093/169647693-4c7607eb-a418-4a79-aaa2-ddefbf906db8.jpg)
> 
> Where should I be unfinished

For android, the original ceserver supports the arm64 debugger.
Debugging is possible by building the cheat engine itself and ceserver.

**Comment by LanBaiCode** [comment]
> ![1](https://user-images.githubusercontent.com/63853093/169647686-6197f8c9-ebe5-4be9-b065-eec3a8f0b2f0.jpg) ![2](https://user-images.githubusercontent.com/63853093/169647693-4c7607eb-a418-4a79-aaa2-ddefbf906db8.jpg)
> 
> Where should I be unfinished

Yes, for Android, it would be better to use native ceserver, no need to use this project

**Comment by yazigegeda** [comment]
I use the source code of cheat engine 7.4 on github to compile ceserver. After execution, this problem still occurs when debugging with cheat engine 7.4

**Comment by yazigegeda** [comment]
> > ![1](https://user-images.githubusercontent.com/63853093/169647686-6197f8c9-ebe5-4be9-b065-eec3a8f0b2f0.jpg) ![2](https://user-images.githubusercontent.com/63853093/169647693-4c7607eb-a418-4a79-aaa2-ddefbf906db8.jpg)
> > Where should I be unfinished
> 
> Yes, for Android, it would be better to use native ceserver, no need to use this project

I use the source code of cheat engine 7.4 on github to compile ceserver. After execution, this problem still occurs when debugging with cheat engine 7.4

**Comment by StevenLyo** [comment]
can I Use this tool on Mac?

**Comment by DoranekoSystems** [comment]
> can I Use this tool on Mac?

The Mac version of cheat engine does not support network mode, so it is not supported at this time.
